Automattic/wp-calypso
# chore(`data-stores`): move `@tanstack/query` based custom hooks to `help-center`

## Proposed Changes

The intent for the `@automattic/data-stores` package is to hold `@wordpress/data`-based stores. We therefore move custom query and mutation hooks (based on `@tanstack/query`) to a more fitting place. It's mostly `@automattic/help-center` related hooks which we move to a `data/` directory inside that package.

Please note there are a few remaining hooks which require a bit more thought on where to put them and/or are otherwise being worked on.

## Testing Instructions

Can and should be reviewed commit by commit. The first commit moves any hooks from `support-queries/` in one go where as all other commits move either isolated hooks or hooks which depend on each other in some way.

In general, a smoke test of the Help Center (on simple and atomic sites, in Calypso as well as wp-admin) is a good start as most of the hooks are related. In more detail:

Verify the flows under "Still need help" work as expected. On the "Public Forums" one (while on a site with a free plan) verify the site analysis runs when providing a different site url than the site you're currently on.

Verify there are no type errors and no usages of the moved hooks where imports aren't updated accordingly.

## Pre-merge Checklist
